backend/api_handler.py

```diff
@@ -207,20 +200,60 @@ def handler(event, context):
                 'body': json.dumps(response),
             }
 
-    success, response = graphql_sync(schema=executable_schema, data=query, context_value=app_context)
-
-    dispose_context()
-    response = json.dumps(response)
-
-    log.info('Lambda Response %s', response)
 
-    return {
-        'statusCode': 200 if success else 400,
-        'headers': {
-            'content-type': 'application/json',
-            'Access-Control-Allow-Origin': '*',
-            'Access-Control-Allow-Headers': '*',
-            'Access-Control-Allow-Methods': '*',
-        },
-        'body': response,
-    }
+def validate_and_block_if_maintenance_window(query, groups):
+    # Logic to block when in maintenance
+    # Check if in some maintenance mode
+    # Check if in maintenance status is not INACTIVE
+    # Check if the user belongs to a 'DAAdministrators' group
+    if config.get_property('modules.maintenance.active'):
+        maintenance_mode = MaintenanceService._get_maintenance_window_mode(engine=ENGINE)
+        maintenance_status = MaintenanceService.get_maintenance_window_status(engine=ENGINE).status
+        isAdmin = TenantPolicyValidationService.is_tenant_admin(groups)
+
+        if (
+            (maintenance_mode == MaintenanceModes.NOACCESS.value)
+            and (maintenance_status is not MaintenanceStatus.INACTIVE.value)
+            and not isAdmin
+        ):
+            if query.get('operationName', '') not in MAINTENANCE_ALLOWED_OPERATIONS:
+                return send_unauthorized_response(
+                    query=query,
+                    message='Access Restricted: data.all is currently undergoing maintenance, and your actions are temporarily blocked.',
+                )
+        elif (
+            (maintenance_mode == MaintenanceModes.READONLY.value)
+            and (maintenance_status is not MaintenanceStatus.INACTIVE.value)
```

Comment:
MaintenanceStatus.INACTIVE is an enum, so you should compare directly with the enum instead of its value.

Reply:
maintenance_status contains the value of the MaintenanceStatus.INACTIVE enum. Let me check if I can get an enum there instead